### PR TITLE
feat: added archive columns and backfill

### PR DESCRIPTION
**Pull request by VB-at-Bis**
Implements #1393

- add archived_at/archived_by columns to application_status table;
- back fill archived_at/archived_by values using ccbc_public.record_version;
- add trigger to application_status table to archive old record when new is added;

### CODE REVIEW

File: db/deploy/util_functions/upsert_archive_trigger.sql [hunk 1]
```diff
@@ -0,0 +1,53 @@
+-- Deploy ccbc:util_functions/upsert_archive_trigger to pg
+-- requires: trigger_function/set_archived
+
+begin;
+
+create or replace function ccbc_private.upsert_archive_trigger(
+  table_schema_name text,
+  table_name text,
+  user_table_schema_name text default 'ccbc_public',
+  user_table_name text default 'ccbc_user'
```
Comment (AntBush) [inline db/deploy/util_functions/upsert_archive_trigger.sql:10]: I notice `user_table_schema_name` and  `user_table_name` isn't used, should probably delete it but won't block approval

File: db/deploy/util_functions/upsert_archive_trigger.sql [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+)
+returns void as $$
+
+declare
+  column_string text;
+  index_string text;
+  comment_string text;
```
Comment (AntBush) [inline db/deploy/util_functions/upsert_archive_trigger.sql:17]: Looks like some of these variables are also unused

**Review by AntBush — APPROVED** [review]
Other than a few minor changes looks great!

File: db/deploy/util_functions/upsert_archive_trigger.sql (hunk 1 above)
Comment (VB-at-Bis) [reply db/deploy/util_functions/upsert_archive_trigger.sql]: indeed  user_table_schema_name and user_table_name isn't used. I deleted them.

File: db/deploy/util_functions/upsert_archive_trigger.sql (hunk 2 above)
Comment (VB-at-Bis) [reply db/deploy/util_functions/upsert_archive_trigger.sql]: Removed

**Pull request merged**